### PR TITLE
unnest endpoint_opts in tests to fix some ci failures

### PR DESCRIPTION
**Pull request by ruslandoga**
This PR should fix at least some of the failing tests. Comments inline.

### CODE REVIEW

File: test/site_encrypt_test.exs [hunk 1]
```diff
@@ -30,12 +30,10 @@ for {input, index} <- Enum.with_index(inputs),
     setup_all do
       start_supervised!({
         TestEndpoint,
-        endpoint_opts: [
```
Comment (ruslandoga) [inline test/site_encrypt_test.exs:33]: This caused opts to be

```elixir
[
  endpoint: SiteEncrypt.Phoenix.Endpoint.Cowboy2Adapter.NativeTest.TestEndpoint,
  endpoint_opts: [
    endpoint_opts: [
      adapter: Phoenix.Endpoint.Cowboy2Adapter,
      http: [port: 40000],
      https: [port: 40001],
      url: [scheme: "https", host: "localhost", port: 40001]
    ]
  ]
]
```

Comment (ruslandoga) [reply test/site_encrypt_test.exs:33]: Not sure though if it's a test issue or a configuration issue.

**Comment by ruslandoga** [comment]
The dialyzer error seems to be a bug in Elixir 1.17.0, fixed in 1.17.1 already.

File: test/site_encrypt_test.exs (hunk 1 above)
Comment (sasa1977) [reply test/site_encrypt_test.exs:33]: It was introduced in a680884f09d0aa6038cfdc297b6cfadfdc2fe0a6. This only affected tests though.

**Comment by sasa1977** [comment]
Thank you!

**Pull request merged**